libs/src/services/dataContracts/userReplicaSetManagerClient.js

```diff
@@ -0,0 +1,161 @@
+   * Callable by user wallet, or any node within the user's replica set
+   * @param {number} userId
+   * @param {number} primary
+   * @param {Array<number>} secondaries
+   */
+  async updateReplicaSet (userId, primary, secondaries) {
+    let existingReplicaSetInfo = await this.getUserReplicaSet(userId)
+    return this._updateReplicaSet(
+      userId,
+      primary,
+      secondaries,
+      existingReplicaSetInfo.primaryId,
+      existingReplicaSetInfo.secondaryIds
+    )
+  }
+
+  /**
+   * Add a new content node to the L2 layer of the protocol
+   * Requires signatures from 3 existing nodes on the UserReplicaSetManager contract
+   * @param {number} cnodeId
+   * @param {Array<string>} cnodeOwnerWallets - [0] = incoming delegateOwnerWallet, [1] = incoming ownerWallet
+   * @param {Array<number>} proposerSpIds
+   * @param {Array<string>} proposerNonces
+   * @param {string} proposer1Sig
+   * @param {string} proposer2Sig
+   * @param {string} proposer3Sig
+   */
+  async addOrUpdateContentNode (
+    cnodeId,
+    cnodeOwnerWallets,
+    proposerSpIds,
+    proposerNonces,
+    proposer1Sig,
+    proposer2Sig,
+    proposer3Sig
+  ) {
+    const contractAddress = await this.getAddress()
+    const method = await this.getMethod(
+      'addOrUpdateContentNode',
+      cnodeId,
+      cnodeOwnerWallets,
+      proposerSpIds,
+      proposerNonces,
+      proposer1Sig,
+      proposer2Sig,
+      proposer3Sig
+    )
+    const tx = await this.web3Manager.sendTransaction(
+      method,
+      this.contractRegistryKey,
+      contractAddress
+    )
+    return tx
+  }
+
+  /**
+   * Generate the relevant data required to propose a new content node
+   * Each incoming node requires 3 distinct signatures in order to be added
+   * This function will be used by content nodes
+   * @param {number} cnodeId
+   * @param {string} cnodeDelegateWallet
+   * @param {number} proposerSpId
+   */
+  async getProposeAddOrUpdateContentNodeRequestData (
+    cnodeId,
+    cnodeDelegateWallet,
+    cnodeOwnerWallet,
+    proposerSpId
+  ) {
+    const chainId = await this.getEthNetId()
+    const contractAddress = await this.getAddress()
+    const nonce = signatureSchemas.getNonce()
+    const signatureData = signatureSchemas.generators.getProposeAddOrUpdateContentNodeRequestData(
+      chainId,
+      contractAddress,
+      cnodeId,
+      cnodeDelegateWallet,
+      cnodeOwnerWallet,
+      proposerSpId,
+      nonce
+    )
+    let sig = await this.web3Manager.signTypedData(signatureData)
+    return {
+      nonce,
+      signatureData,
+      sig
+    }
+  }
+
+  /**
+   * Return the current replica set associated with a given user
+   * @param {number} userId
+   */
+  async getUserReplicaSet (userId) {
+    const method = await this.getMethod('getUserReplicaSet', userId)
+    let currentWallet = this.web3Manager.getWalletAddress()
+    return method.call({ from: currentWallet })
```

Comment:
prob we should wrap this inside a `webManager.sendCall()` function or something but w/e, works for now

Reply:
eh